### ISSUE TITLE
convenience syntax for array construction

### ISSUE THREAD

**Issue by japm48** [numpy/numpy#12233]
Apparently `np.matrix` is getting deprecated, and my only used feature from it is the Matlab-style string initializer (e.g. `np.matrix('1 2; 3 4')`).

`np.array('1 2; 3 4')` apparently builds a character array, so a different syntax should be used.

Suggestion: `np.arr()` (analogous to `np.mat()`).

### Reproducing code example:

```python
import numpy as np
A = np.matrix('1 2; 3 4')
B = np.array('1 2; 3 4') # Wrong expected value
```

**Comment by japm48** [comment]
@ilayn Actually, I have already done exactly what you suggest before opening the issue (thanks for the effort, anyways).
I am just suggesting having this functionality built in arrays by default.

**Comment by eric-wieser** [comment]
I don't think this functionality makes sense as part of numpy. We should be avoiding designing a mini-language within string literals as much as possible.

Why is `array('1 2;3 4')` more convenient than `array([[1,2],[3,4]])`? What if you want `array([[foo(),2],[3,4]])`? The convenience syntax turns out to be inconvenient in all but an extremely narrow set of cases, and even in those cases the only real convenience is a resemblance to matlab.
